RecoPixelVertexing/PixelTriplets/src/CACut.h

```diff
@@ -0,0 +1,175 @@
+        if (thisCACut.layerIds.back() == -1) {
+          foundAllLayerIds = false;
+          edm::LogWarning("Configuration")
+              << "Layer name '" << layerName << "' not found in the CAGraph. Please check CACuts parameter set.";
+        }
+      }
+
+      // Cut
+      thisCACut.cutValue = thisTriplet.cutValue;
+      thisCACut.hasValueByInnerLayerId = false;
+    }
+
+    setCutValuesByInnerLayerIds();
+  }
+
+  void setCutValuesByInnerLayerIds() {
+    for (auto &thisTriplet : valuesByLayerIds_) {
+      if (thisTriplet.hasValueByInnerLayerId)
+        continue;
+      auto it = std::find(thisTriplet.layerIds.begin(), thisTriplet.layerIds.end(), -1);
+      if (it != thisTriplet.layerIds.end())
+        continue;
+
+      bool foundOuterDoublet = false;
+
+      for (auto &thisOuterDoublet : valuesByInnerLayerIds_) {
+        if (thisOuterDoublet.outerDoubletIds[0] == thisTriplet.layerIds[1] &&
+            thisOuterDoublet.outerDoubletIds[1] == thisTriplet.layerIds[2]) {
+          thisOuterDoublet.innerLayerIds.emplace_back(thisTriplet.layerIds[0]);
+          thisOuterDoublet.cutValues.emplace_back(thisTriplet.cutValue);
+          foundOuterDoublet = true;
+          break;
+        }
+      }
+
+      if (!foundOuterDoublet) {
+        valuesByInnerLayerIds_.emplace_back(defaultCut_);
+        auto &newOuterDoublet = valuesByInnerLayerIds_.back();
+
+        newOuterDoublet.outerDoubletIds.emplace_back(thisTriplet.layerIds[1]);
+        newOuterDoublet.outerDoubletIds.emplace_back(thisTriplet.layerIds[2]);
+        newOuterDoublet.innerLayerIds.emplace_back(thisTriplet.layerIds[0]);
+        newOuterDoublet.cutValues.emplace_back(thisTriplet.cutValue);
+      }
+
+      thisTriplet.hasValueByInnerLayerId = true;
+    }
+  }
+
+  class CAValuesByInnerLayerIds {
+  public:
+    explicit CAValuesByInnerLayerIds(float cut) : defaultCut_(cut) {}
+
+    float at(int layerId) const {
+      for (size_t thisLayer = 0; thisLayer < innerLayerIds.size(); thisLayer++) {
+        if (innerLayerIds.at(thisLayer) == layerId)
+          return cutValues.at(thisLayer);
+      }
+
+      return defaultCut_;
+    }
+
+    std::vector<int> outerDoubletIds;
+    std::vector<int> innerLayerIds;
+    std::vector<float> cutValues;
+
+  private:
+    double defaultCut_;
+  };
+
+  CAValuesByInnerLayerIds getCutsByInnerLayer(int layerIds1, int layerIds2) const {
+    for (const auto &thisCut : valuesByInnerLayerIds_) {
+      if (thisCut.outerDoubletIds[0] == layerIds1 && thisCut.outerDoubletIds[1] == layerIds2) {
+        return thisCut;
+      }
+    }
+
+    CAValuesByInnerLayerIds emptyCutsByInnerLayer(defaultCut_);
+    return emptyCutsByInnerLayer;
+  }
+
+private:
+  class CAValueByTripletName {
+  public:
+    std::string tripletName;
+    float cutValue;
+  };
+
+  class CAValueByLayerIds {
+  public:
+    std::vector<int> layerIds;
+    float cutValue;
+    bool hasValueByInnerLayerId;
+  };
+
+public:
+  bool useCACuts;
```

Comment:
Trailing underscore for local variables, as you already to for the private ones.
By the way: why do those two need to be public?

Reply:
However, ```useCACuts``` and ```foundAllLayerIds``` are  only set here, but never used: what are they expected to be used for? Shouldn't they just get removed?

Reply:
They are used in setCutValuesByLayerIds to skip the layer name -> layer id mapping when not needed. Would you suggest to do this differently?